apps/api/src/app/events/usecases/message-matcher/message-matcher.usecase.ts

```diff
@@ -13,21 +13,31 @@ import {
   FilterPartTypeEnum,
   ICondition,
   TimeOperatorEnum,
+  ChannelTypeEnum,
+  IPreviousStepFilterPart,
+  PreviousStepTypeEnum,
 } from '@novu/shared';
 import { ExecutionDetailsSourceEnum, ExecutionDetailsStatusEnum } from '@novu/shared';
-import { SubscriberEntity, EnvironmentRepository, SubscriberRepository, StepFilter } from '@novu/dal';
+import {
+  SubscriberEntity,
+  EnvironmentRepository,
+  SubscriberRepository,
+  StepFilter,
+  ExecutionDetailsRepository,
+  MessageRepository,
+  JobRepository,
+  MessageEntity,
+} from '@novu/dal';
 
 import { IFilterVariables } from './types';
 import { FilterProcessingDetails } from './filter-processing-details';
-
-import {
-  CreateExecutionDetails,
-  CreateExecutionDetailsCommand,
-} from '../../../execution-details/usecases/create-execution-details';
+import { CreateExecutionDetails } from '../../../execution-details/usecases/create-execution-details';
 import { SendMessageCommand } from '../send-message/send-message.command';
 import { EXCEPTION_MESSAGE_ON_WEBHOOK_FILTER } from '../../../shared/constants';
-import { DetailEnum } from '../../../execution-details/types';
+import { CreateExecutionDetailsCommand } from '../../../execution-details/usecases/create-execution-details/create-execution-details.command';
+import { EmailEventStatusEnum } from '@novu/stateless';
 import { ApiException } from '../../../shared/exceptions/api.exception';
+import { DetailEnum } from '../../../execution-details/types';
```

Comment:
Should we move the enum to shared to avoid weird relative pathing between modules? Considering how reused is the execution details module.

Reply:
Do not like to do it in this pr will create tech debt about it...